eng/pipelines/variables/common.yml

```diff
@@ -19,5 +19,11 @@ variables:
 - name: testScriptPath
   value: ""
 - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
-    - name: build.imageBuilderDockerRunExtraOptions
-      value: -e DOCKER_REPO=$(acr.server)/$(stagingRepoPrefix)dotnet-buildtools/prereqs
+  - name: build.imageBuilderDockerRunExtraOptions
+    value: -e DOCKER_REPO=$(acr.server)/$(stagingRepoPrefix)dotnet-buildtools/prereqs
+- ${{ if eq(variables['Build.SourceBranch'], 'refs/heads/production') }}:
+  - name: LatestTag
+    value: 'latest'
+- else:
+  - name: LatestTag
+    value: 'staging'
```

Comment:
Setting variables in the pipeline doesn't cause them to automatically be flowed into the build. Instead, there's a `manifestVariables` variable which needs to be set, with a multi-value `--var` option:

https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/534c91f7f0644c7a2eeb58b95fbf0274266c083c/eng/pipelines/variables/common.yml#L13-L14

So you would include `--var LatestTag latest` on this line, for example. That will allow you to reference it in the manifest as a variable.

In addition, you'll want to update the local build script to default this with an appropriate value:

https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/534c91f7f0644c7a2eeb58b95fbf0274266c083c/build.ps1#L10

Reply:
Sorry I wasn't clear on my earlier comment. All that needs to be done is to change this: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/534c91f7f0644c7a2eeb58b95fbf0274266c083c/eng/pipelines/variables/common.yml#L14

To this:

```
value: --var UniqueId=$(sourceBuildId) --var LatestTag=latest
```

or this:

```
value: --var UniqueId=$(sourceBuildId) --var LatestTag=staging
```

based on the source branch condition.

Once you've done that, the other pipeline changes are unnecessary.